src/Stratis.Bitcoin.Features.Consensus.Tests/PosConsensusRulesRegistrationTests.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using Stratis.Bitcoin.Features.Consensus.Rules;
+using Stratis.Bitcoin.Features.Consensus.Rules.CommonRules;
+using Xunit;
+
+namespace Stratis.Bitcoin.Features.Consensus.Tests
+{
+    public class PosConsensusRulesRegistrationTests
+    {
+        private readonly IEnumerable<ConsensusRule> rules;
+
+        public PosConsensusRulesRegistrationTests()
+        {
+            this.rules = new FullNodeBuilderConsensusExtension.PosConsensusRulesRegistration().GetRules();
+        }
+
+        [Fact]
+        public void GetRules_ForPOS_ReturnsListOfRegisteredPowRules()
+        {
+            this.rules.Count().Should().Be(22);
+
+            this.rules.ElementAt(0).Should().BeOfType<BlockHeaderRule>();
```

Comment:
ah interesting this is a good idea.

Reply:
Is it? I'm kind of missing the purpose of this test.Can you explain?

Reply:
The problem is that this test does not guarantee the correct order. It is just long-winded way how to copy the original list.

Reply:
Why won't it guarantee the order?

Reply:
because it is only a copy of the list, it has no logic to know whether the main code order was correct

Reply:
basically it is a test that is fit to the code, so that current code passes it, as such it is very much not testing anything as if the current order was wrong, you would not get test failed from it

Reply:
It is testing that the rules have been correctly ordered on the Registration class, if someone breaks the order by mistake this will fail (its already better then nothing) testing on the engine itself can also be a good idea.

However we have to remember this will be network specific and later rules will move to the Network.Consensus class.

I am fine with this as is right now as it holds some value.

Reply:
In my opinion it is not testing that the rules have been correctly ordered. I don't mind having this test inside, it is just not doing what you think it is doing.

Reply:
You keep saying that is it not testing they are correctly ordered, but it is testing they are correctly ordered. Arguably we could move this onto the Regsitration itself and check that the IConsensusRules (of pow and pos) each have their Rules correctly set.

There is no proof within this test that the system in general even chooses to use this list of rules, however it does prove that it if does that the rules are set in the correct order.

Can you suggest something better instead of this?

Reply:
I still can't see how is it testing the correct order. In my opinion it is only testing that the list in one file is in the same order as list in another file, nothing else. So the main value of this test is to prevent changing the main code without thinking twice. And that is because you will have failed test and maybe you will think twice. Quite likely so, you will just fix the test list in the same way you changed the main code list, therefore this test is quite weak in achieving even that.

I don't think the current design allows you to test the correct order. Some improvements could be made to have some weaker guarantees between some of the rules, but the full correct ordering is probably difficult to achieve this way. It makes much better sense to create a comment that argues about ordering and explains why the ordering selected as it is.

Reply:
I cannot accept that a comment is an improvement over a unit test.

Reply:
Created https://github.com/stratisproject/StratisBitcoinFullNode/issues/1435  to reflect this discussion for improvement